src/csharp/Extensions/Microsoft.Spark.Extensions.Hyperspace.E2ETest/HyperspaceTests.cs

```diff
@@ -0,0 +1,160 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.Spark.E2ETest.Utils;
+using Microsoft.Spark.Extensions.Hyperspace.Index;
+using Microsoft.Spark.Sql;
+using Xunit;
+
+namespace Microsoft.Spark.Extensions.Hyperspace.E2ETest
+{
+    /// <summary>
+    /// Test suite for Hyperspace index management APIs.
+    /// </summary>
+    [Collection(Constants.HyperspaceTestContainerName)]
+    public class HyperspaceTests : IDisposable
+    {
+        private readonly HyperspaceFixture _fixture;
+        private readonly SparkSession _spark;
+        private readonly Hyperspace _hyperspace;
+
+        // Fields needed for sample DataFrame.
+        private readonly DataFrame _sampleDataFrame;
+        private readonly string _sampleIndexName;
+        private readonly IndexConfig _sampleIndexConfig;
+
+        public HyperspaceTests(HyperspaceFixture fixture)
+        {
+            _fixture = fixture;
+            _spark = fixture.SparkFixture.Spark;
+            _hyperspace = fixture.Hyperspace;
+
+            // Make sure that any leftover indexes are removed before starting a test.
+            DeleteHyperspaceSystemData();
+
+            _sampleDataFrame = _spark.Read()
+                .Option("header", true)
+                .Csv("Resources\\sample-data.csv");
+            _sampleIndexName = "sample_dataframe";
+            _sampleIndexConfig = new IndexConfig(
+                _sampleIndexName,
+                new List<string> { "c3" },
+                new List<string> { "c1" });
+            // Create a simple index.
+            _spark.EnableHyperspace();
```

Comment:
you are not running any queries utilizing the index, so no need to call this. Same for `DisableHyperspace` in `Dispose`.

Reply:
Removed.